integration/mediation-tests/tests-other/src/test/java/org/wso2/carbon/esb/connector/test/MediationLibraryServiceTestCase.java

```diff
@@ -0,0 +1,310 @@
+        waitUntilLibAvailable(AMAZON_CONNECTOR_ZIP);
+        Assert.assertFalse(checkAvailabilityInImports(AMAZON_CONNECTOR_LIB_QNAME),
+                           "Connector is already enabled at the deployment.");
+
+        //enable connector
+        updateConnectorStatus(AMAZON_CONNECTOR_LIB_QNAME, AMAZON_LIB_NAME, PACKAGE_NAME, ENABLED);
+        Thread.sleep(5000);
+        Assert.assertTrue(checkAvailabilityInImports(AMAZON_CONNECTOR_LIB_QNAME),
+                          "Connector is still in the disable state after enable action.");
+
+        //remove connector
+        deleteLibrary(AMAZON_CONNECTOR_LIB_QNAME);
+        Thread.sleep(5000);
+        Assert.assertFalse(checkAvailabilityInImports(AMAZON_CONNECTOR_LIB_QNAME),
+                           "Connector is still available in the imports list after remove action.");
+    }
+
+    /**
+     * Test for functionality related to imports including add, get and delete.
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.esb", description = "Test add, get and delete functionality for imports.")
+    public void importsFunctionalityTest() throws Exception {
+
+        String expectedImport = "<import xmlns=\"http://ws.apache.org/ns/synapse\" name=\"hello\" package=\"org.wso2" +
+                ".carbon.connector\" status=\"enabled\"/>";
+
+        //add import
+        Assert.assertFalse(checkAvailabilityInImports(TEMP_CONNECTOR_LIB_QNAME),
+                           "Import is already available before add action.");
+
+        addImport(TEMP_LIB_NAME, PACKAGE_NAME);
+        Assert.assertTrue(checkAvailabilityInImports(TEMP_CONNECTOR_LIB_QNAME),
+                          "Import is not available after add action.");
+
+        //get import
+        Assert.assertEquals(getImport(HELLO_CONNECTOR_LIB_QNAME), expectedImport,
+                            "Received synapse configuration for hello-connector is incorrect.");
+
+        try {
+            getImport(INVALID_CONNECTOR);
+            Assert.fail("Get import fails for an invalid connector.");
+        } catch (AxisFault e) {
+            Assert.assertEquals(e.getMessage(), "Library Import null does not exist");
+        }
+
+        //delete import
+        deleteImport(TEMP_CONNECTOR_LIB_QNAME);
+        Assert.assertFalse(checkAvailabilityInImports(TEMP_CONNECTOR_LIB_QNAME),
+                           "Import is still available after delete action.");
+    }
+
+    /**
+     * Test for library information.
+     *
+     * @throws Exception
+     */
+    @Test(groups = "wso2.esb", description = "Test retrieving library information.")
+    public void libraryInfoTest() throws Exception {
+
+        LibraryInfo[] allLibraryInfo = getAllLibraryInfo();
+        LibraryInfo helloLibraryInfo = null;
+
+        if (null != allLibraryInfo && allLibraryInfo.length > 0) {
+            for (int i = 0; i < allLibraryInfo.length; i++) {
+                if (allLibraryInfo[i].getLibName().equals(HELLO_LIB_NAME)) {
```

Comment:
Shall we swap the two params to be safe. if (HELLO_LIB_NAME.equals(allLibraryInfo[i].getLibName())) ??

Reply:
Updated.